test/unittests/classification/test_confusion_matrix.py

```diff
@@ -17,189 +17,352 @@
+    @pytest.mark.parametrize("ddp", [True, False])
+    def test_multilabel_confusion_matrix(self, input, ddp, normalize, ignore_index):
+        preds, target = input
+        if ignore_index is not None:
+            target = inject_ignore_index(target, ignore_index)
+        self.run_class_metric_test(
+            ddp=ddp,
+            preds=preds,
+            target=target,
+            metric_class=MultilabelConfusionMatrix,
+            sk_metric=partial(_sk_confusion_matrix_multilabel, normalize=normalize, ignore_index=ignore_index),
+            metric_args={
+                "num_labels": NUM_CLASSES,
                 "normalize": normalize,
-                "multilabel": multilabel,
+                "ignore_index": ignore_index,
             },
         )
 
-    def test_confusion_matrix_differentiability(self, normalize, preds, target, sk_metric, num_classes, multilabel):
-        self.run_differentiability_test(
+    def test_multilabel_confusion_matrix_functional(self, input, normalize, ignore_index):
+        preds, target = input
+        if ignore_index is not None:
+            target = inject_ignore_index(target, ignore_index)
+        self.run_functional_metric_test(
             preds=preds,
             target=target,
-            metric_module=ConfusionMatrix,
-            metric_functional=confusion_matrix,
+            metric_functional=multilabel_confusion_matrix,
+            sk_metric=partial(_sk_confusion_matrix_multilabel, normalize=normalize, ignore_index=ignore_index),
             metric_args={
-                "num_classes": num_classes,
-                "threshold": THRESHOLD,
+                "num_labels": NUM_CLASSES,
                 "normalize": normalize,
-                "multilabel": multilabel,
+                "ignore_index": ignore_index,
             },
         )
 
 
-def test_warning_on_nan(tmpdir):
+def test_warning_on_nan():
     preds = torch.randint(3, size=(20,))
     target = torch.randint(3, size=(20,))
 
     with pytest.warns(
         UserWarning,
-        match=".* nan values found in confusion matrix have been replaced with zeros.",
+        match=".* NaN values found in confusion matrix have been replaced with zeros.",
     ):
-        confusion_matrix(preds, target, num_classes=5, normalize="true")
-
-
-@pytest.mark.parametrize(
-    "metric_args",
-    [
-        {"num_classes": 1, "normalize": "true"},
-        {"num_classes": 1, "normalize": "pred"},
-        {"num_classes": 1, "normalize": "all"},
-        {"num_classes": 1, "normalize": "none"},
-        {"num_classes": 1, "normalize": None},
-    ],
-)
-def test_provide_superclass_kwargs(metric_args: Dict[str, Any]):
-    """Test instantiating subclasses with superclass arguments as kwargs."""
-    JaccardIndex(**metric_args)
+        multiclass_confusion_matrix(preds, target, num_classes=5, normalize="true")
+
+
+# -------------------------- Old stuff --------------------------
```

Comment:
Let's delete it or do you want to use it later?

Reply:
let's not delete it for now. We want to reuse it later on, so we do need some reference. Let's clean up in the end